### ISSUE TITLE
No foreign keys

### ISSUE THREAD

**Issue by alpapan** [rainlab/user-plugin#365]
Hello
I've been going through the schema of october cms/rainlab plugins and it seems that foreign keys are missing. Even though this is known for almost 3 years (https://github.com/rainlab/user-plugin/issues/151#issuecomment-188945870).

Is there are a good reason for this? If not then it is a serious DBA flaw...

**Comment by alpapan** [comment]
First, I tried to include FK statements when tables were being created but I couldn't figure out the right order. It seems the tables are not created in a fashion I can understand. Therefore, I think the FKs should be added last (in my plugin i always make sure the tables are created with FKs and in the right order). Anyhow, the solution would be to add a migration at the end of the October install.

Second, I found it quite hard to identify the correct key relationships (and I am pretty sure only the developer should be doing this) but I'm attaching a migration that affects 
- the backend 
- front end users.

It also has an entry I found in blog but I've commented it out as this is about the User plugin.

DISCLAIMER: I didn't look for all foreign keys, it would be impossible for someone who doesn't have deep knowledge of the models to find them correctly. However, the attached migration works as a template and finds most user /role FKs

Note it is namespaced for a plugin i've been writing so that will have to change too. I will not be doing a PR but if someone is interested, here it is.

[patch_october.zip](https://github.com/rainlab/user-plugin/files/2755355/patch_october.zip)

thank you
a
